### PR TITLE
Resolve warnings and errors that come with updating to node 16

### LINKED ISSUE #438: Move to NodeJS 16
From comments on Slack I gather that @mook-as and @ericpromislow are using NodeJS 15.* for current development.

NodeJS 15 has been end-of-lifed since June 1st 2021: [NodeJS Releases](https://nodejs.org/en/about/releases/).

RD for Silicon will require NodeJS 16 as well: https://github.com/rancher-sandbox/rancher-desktop/issues/16#issuecomment-881741749

### PR DESCRIPTION
**Pull request by rak-phillip**
This is a series of refactorings for updating to node v16. Tested with node v16.13.1

closes #438

### CODE REVIEW

**Comment by rak-phillip** [comment]
A lot of these changes are related to running `npm update` after moving to node v16 to resolve some `node-gyp` issues that I encountered in my environment. After taking a step back and thinking about this change, I could probably attempt to focus on updating only the packages that have the `node-gyp` dependency if this blanket update of our dependencies are not desired.

I see that we also have #1161 that seems to describe some related behavior to what I was seeing. We might want to check to see if this is resolved as a result of this change.

**Review by ericpromislow — CHANGES_REQUESTED** [review]
One of the `String.substring` uses needs fixing.

And as Phillip and I communicated in slack, I initially didn't see a need for these changes, even after upgrading from node 16.9.1 to 16.13.1. Then I cleared `./node_modules/.cache/babel-loader/` and `lint` showed about 30 problems.

File: src/k8s-engine/lima.ts [hunk 1]
```diff
@@ -655,7 +655,7 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
 
   private static calcRandomTag(desiredLength: number) {
     // quicker to use Math.random() than pull in all the dependencies utils/string:randomStr wants
-    return Math.random().toString().substr(2, desiredLength);
+    return Math.random().toString().substring(2, desiredLength);
```
[the remark this answers is not in the document]
Comment (rak-phillip) [reply src/k8s-engine/lima.ts]: Good catch, thanks for the heads up! Fixed

**Review by ericpromislow — APPROVED** [review]
LGTM

**Pull request merged**